### PR TITLE
fix: `highWaterMark` bug @ v5

### PR DESCRIPTION
**Pull request by darcyclarke**

### CODE REVIEW

**Comment by isaacs** [comment]
Travis errors are spurious npm v3 problems.  Ignore.

File: test/cache.js [hunk 1]
```diff
@@ -895,9 +895,6 @@ test('supports matching using Vary header', t => {
   })
 })
 
-test('supports range caching (partial requests)')
```
Comment (isaacs) [inline test/cache.js]: These todo tests are probably best to leave around?  I know they're noisy, but they're helpful to ensure we don't forget about important use cases when we do get time to bump up the test coverage in this repo.  (Though, I guess to be fair they are less useful in v5 than in latest, so removing on this branch is probably fine.)

**Comment by isaacs** [comment]
Looks good to me, but we don't typically bump the version in a PR?  Doesn't `npm run release` take care of all of that?

**Comment by darcyclarke** [comment]
@isaacs fixed up the PR to be inline with whatever the standard was. Can do a patch release once it's in `v5` & am opened a similar PR for `v6` [here](https://github.com/npm/make-fetch-happen/pull/11).

Will rock the boat later with changes to tests, docs etc. 😉

**Comment by claudiahdz** [comment]
LGTM!

**Pull request merged**